yarn-project/world-state/src/native/native_bench.test.ts

```diff
@@ -0,0 +1,292 @@
+        leaves.push(l2Block.body.txEffects[0].publicDataWrites.map(write => write.toBuffer()));
+      } else if (treeId === MerkleTreeId.NULLIFIER_TREE) {
+        const nullifiersPadded = padArrayEnd(l2Block.body.txEffects[0].nullifiers, Fr.ZERO, MAX_NULLIFIERS_PER_TX).map(
+          nullifier => nullifier.toBuffer(),
+        );
+        leaves.push(nullifiersPadded);
+      } else {
+        const noteHashesPadded = padArrayEnd(l2Block.body.txEffects[0].noteHashes, Fr.ZERO, MAX_NOTE_HASHES_PER_TX);
+        leaves.push(noteHashesPadded);
+      }
+    }
+    const fork = await worldState.fork();
+    const startTime = performance.now();
+
+    if (treeId === MerkleTreeId.NOTE_HASH_TREE) {
+      for (const leafSet of leaves) {
+        await fork.appendLeaves(MerkleTreeId.NOTE_HASH_TREE, leafSet as Fr[]);
+      }
+    } else if (insertionType === InsertionType.BATCH) {
+      for (const leafSet of leaves) {
+        await fork.batchInsert(treeId as IndexedTreeId, leafSet as Buffer[], 0);
+      }
+    } else {
+      for (const leafSet of leaves) {
+        await fork.sequentialInsert(treeId as IndexedTreeId, leafSet as Buffer[]);
+      }
+    }
+    const endTime = performance.now();
+    const avgTime = (endTime - startTime) / numBlocks;
+    return avgTime;
+  };
+
+  const runDataRetrievalByIndex = async (
+    numRetrievals: number,
+    treeId: MerkleTreeId,
+    worldState: NativeWorldStateService,
+    op: (treeId: MerkleTreeId, index: bigint, fork: MerkleTreeReadOperations) => Promise<any>,
+  ) => {
+    const fork = await worldState.fork();
+    const treeInfo = await fork.getTreeInfo(treeId);
+    const indices = Array.from({ length: numRetrievals }, () =>
+      Math.floor(Math.random() * Number(treeInfo.size - 1n)),
+    ).map(m => BigInt(m));
+
+    const startTime = performance.now();
+
+    for (const index of indices) {
+      await op(treeId, index, fork);
+    }
+
+    const endTime = performance.now();
+    const avgTime = (endTime - startTime) / numRetrievals;
+    return avgTime;
+  };
+
+  const runDataRetrievalByValue = async (
+    numRetrievals: number,
+    treeId: MerkleTreeId,
+    worldState: NativeWorldStateService,
+    values: (Buffer | Fr)[],
+    op: (treeId: MerkleTreeId, value: Buffer | Fr, fork: MerkleTreeReadOperations) => Promise<any>,
+  ) => {
+    const fork = await worldState.fork();
+    const treeInfo = await fork.getTreeInfo(treeId);
+    const indices = Array.from({ length: numRetrievals }, () =>
+      Math.floor(Math.random() * Number(treeInfo.size - 1n)),
+    ).map(m => BigInt(m));
+
+    const startTime = performance.now();
+
+    for (const index of indices) {
+      const value = values[Number(index)];
+      await op(treeId, value, fork);
+    }
+
+    const endTime = performance.now();
+    const avgTime = (endTime - startTime) / numRetrievals;
+    return avgTime;
+  };
+
+  it.each([
+    [36, 1, 2],
+    [36, 8, 2],
+    [36, 64, 2],
+    [360, 8, 1],
```

Comment:
should we test 10tps and big txs too?

Reply:
The reason I haven't is speed. It's pretty slow to run these block sync tests